### PR TITLE
Add support for passing --version flag to m2setup.sh

### LINKED ISSUE #75: Add ability to specify project name via m2setup.sh
Currently the only way to install Magento 2.1 RC using `m2setup.sh` is to do a normal install and then upgrade it to 2.1 using "Upgrade using the command line" instructions on this page: http://devdocs.magento.com/guides/v2.0/release-notes/ReleaseNotes2.1_RC1CE.html

It would be ideal to be able to pass a flag to `m2setup.sh` to use a custom Composer project name (e.g., `magento/project-community-edition=2.1.0-rc1`), per http://devdocs.magento.com/guides/v2.0/release-notes/ReleaseNotes2.1_RC1CE.html#binstall-using-composerb

### PR DESCRIPTION
**Pull request by erikhansen**
Resolves #75

### CODE REVIEW

File: vagrant/guest/bin/m2setup.sh [hunk 1]
```diff
@@ -278,9 +286,14 @@ function install_from_packages {
...
         fi
+        
+        proj_version=""
+        if [[ $PROJ_VERSION ]]; then
+            proj_version=$PROJ_VERSION
+        fi
 
         composer create-project $NOISE_LEVEL --no-interaction --prefer-dist \
-            --repository-url=https://repo.magento.com/ $package_name $INSTALL_DIR
+            --repository-url=https://repo.magento.com/ $package_name $INSTALL_DIR $proj_version
```
Comment (davidalger) [inline vagrant/guest/bin/m2setup.sh]: This should be quoted since it's a user-inputed value.

File: vagrant/guest/bin/m2setup.sh [hunk 2]
```diff
@@ -278,9 +286,14 @@ function install_from_packages {
         if [[ $ENTERPRISE ]]; then
             package_name="magento/project-enterprise-edition"
         fi
+        
+        proj_version=""
+        if [[ $PROJ_VERSION ]]; then
+            proj_version=$PROJ_VERSION
```
Comment (davidalger) [inline vagrant/guest/bin/m2setup.sh]: Not really necessary, but ok…

File: vagrant/guest/bin/m2setup.sh [hunk 3]
```diff
@@ -76,6 +76,13 @@ for arg in "$@"; do
                 exit -1
             fi
             ;;
+        --proj-version=*)
+            PROJ_VERSION="${arg#*=}"
+            if [[ ! "$PROJ_VERSION" =~ ^[a-z0-9\.-]+$ ]]; then
```
Comment (davidalger) [inline vagrant/guest/bin/m2setup.sh]: Can you add an `*` to the allowed character list as well? Ex: 2.0.* can be used to install latest 2.0 patch release. If we have to use this after 2.1 becomes the latest stable release, having the wildcard allowed will be very handy.

**Comment by davidalger** [comment]
@erikhansen Added a couple comments. If you can address those, I'll get this merged in short order.

File: vagrant/guest/bin/m2setup.sh (hunk 1 above)
Comment (erikhansen) [inline vagrant/guest/bin/m2setup.sh]: Good point.

File: vagrant/guest/bin/m2setup.sh (hunk 2 above)
Comment (erikhansen) [inline vagrant/guest/bin/m2setup.sh]: Yeah, I'll remove this.

File: vagrant/guest/bin/m2setup.sh (hunk 3 above)
Comment (erikhansen) [inline vagrant/guest/bin/m2setup.sh]: I questioned whether such a strict validation was necessary. What do you think about just changing the regex to `^.+$`

Comment (davidalger) [inline vagrant/guest/bin/m2setup.sh]: That sounds fine

**Comment by erikhansen** [comment]
@davidalger I regarded your feedback in my latest commit. Screenshot validating that latest changes passed the three test cases I tested against:

image

**Pull request merged**